go.mod

```diff
@@ -1,9 +1,16 @@
 module github.com/peterbourgon/unixtransport
 
-go 1.16
+go 1.20
```

Comment:
Is this strictly necessary?

Reply:
Strictly speaking, versions in go.mod map to capabilities, not Go releases, so stuff like EOL doesn't really matter.

Reply:
Does this PR fix any specific problem?

Reply:
> Strictly speaking, versions in go.mod map to capabilities, not Go releases, so stuff like EOL doesn't really matter.

Since they changed this up in 1.22, I am not sure anymore. It's a bit of a mess.



> Does this PR fix any specific problem?

Yeah, when using 1.16 or below, the unused dependencies are not removed from go.sum and clutter up depending projects.

Reply:
> Yeah, when using 1.16 or below, the unused dependencies are not removed from go.sum and clutter up depending projects.

go.sum isn't a lock file, it's more like an append-only checksum file. It's not used to calculate or define dependencies, it's only used to verify the integrity of downloaded dependencies. So not really anything that can get "cluttered" 😇 

Absent specific e.g. CVEs or bug fixes in the DNS library, I'll pass on this. But thanks for the contribution!